arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -0,0 +1,562 @@
+    if (validationRes.ok()) {
+      return std::make_shared<arangodb::iresearch::AqlAnalyzer>(options);
+    } else {
+      LOG_TOPIC("f775e", WARN, arangodb::iresearch::TOPIC)
+          << "error validating calculation query: " << validationRes.errorMessage();
+    }
+  }
+  return nullptr;
+}
+}  // namespace
+
+namespace arangodb {
+namespace iresearch {
+
+/*static*/ bool AqlAnalyzer::normalize_vpack(const irs::string_ref& args, std::string& out) {
+  auto const slice = arangodb::iresearch::slice(args);
+  VPackBuilder builder;
+  if (normalize_slice(slice, builder)) {
+    out.resize(builder.slice().byteSize());
+    std::memcpy(&out[0], builder.slice().begin(), out.size());
+    return true;
+  }
+  return false;
+}
+
+/*static*/ bool AqlAnalyzer::normalize_json(const irs::string_ref& args,
+                                                    std::string& out) {
+  auto src = VPackParser::fromJson(args);
+  VPackBuilder builder;
+  if (normalize_slice(src->slice(), builder)) {
+    out = builder.toString();
+    return true;
+  }
+  return false;
+}
+
+/*static*/ irs::analysis::analyzer::ptr AqlAnalyzer::make_vpack(irs::string_ref const& args) {
+  auto const slice = arangodb::iresearch::slice(args);
+  return make_slice(slice);
+}
+
+
+/*static*/ irs::analysis::analyzer::ptr AqlAnalyzer::make_json(irs::string_ref const& args) {
+  auto builder = VPackParser::fromJson(args);
+  return make_slice(builder->slice());
+}
+
+AqlAnalyzer::AqlAnalyzer(Options const& options)
+  : irs::analysis::analyzer(irs::type<AqlAnalyzer>::get()),
+    _options(options),
+    _query(new CalculationQueryContext(arangodb::DatabaseFeature::getCalculationVocbase())),
+    _itemBlockManager(&_resourceMonitor, SerializationFormat::SHADOWROWS), 
+    _engine(0, *_query, _itemBlockManager,
+            SerializationFormat::SHADOWROWS, nullptr) {
+  TRI_ASSERT(validateQuery(_options.queryString,
+                           arangodb::DatabaseFeature::getCalculationVocbase())
+                 .ok());
+}
+
+bool AqlAnalyzer::next() {
+  do {
+    if (_queryResults != nullptr) {
+      while (_queryResults->numRows() > _resultRowIdx) {
+        AqlValue const& value =
+            _queryResults->getValueReference(_resultRowIdx++, _engine.resultRegister());
+        if (value.isString() || (value.isNull(true) && _options.keepNull)) {
+          if (value.isString()) {
+            _term.value = arangodb::iresearch::getBytesRef(value.slice());
+          } else {
+            _term.value = irs::bytes_ref::EMPTY;
+          }
+          _inc.value = _nextIncVal;
+          _nextIncVal = !_options.collapsePositions;
+          return true;
+        }
+      }
+    }
+    if (_executionState == ExecutionState::HASMORE) {
+      _executionState = ExecutionState::DONE;  // set to done to terminate in case of exception
+      _resultRowIdx = 0;
+      _queryResults = nullptr;
+      try {
+        std::tie(_executionState, _queryResults) = _engine.getSome(_options.batchSize);
```

Comment:
Please use `_engine.execute()` instead of getSome.
Get some is the old API that we just keep around for backwards compatibility and will remove it soon.
Sorry we forgot to add a deprecated Note there.